src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -87,12 +87,15 @@ public void Abort()
                 Socket connectedSocket = await ConnectSocketAsync(uri.Host, uri.Port, cancellationToken).ConfigureAwait(false);
                 Stream stream = new AsyncEventArgsNetworkStream(connectedSocket);
 
+                // Honor Host header if set.
+                string hostHeader = options.RequestHeaders[HttpKnownHeaderNames.Host];
```

Comment:
@davidsh, it looks like on .NET Framework, calling SetRequestHeader with Host throws an exception, but on .NET Core it's allowed.  I assume that's by design?